corehq/middleware.py

```diff
@@ -120,8 +116,7 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         secure_session = request.session.get('secure_session')
         now = datetime.datetime.utcnow()
 
-        if hasattr(request, 'domain') and \
-                self._session_insecure_and_domain_requires_secure_session(secure_session, request.domain):
+        if not secure_session and self._user_requires_secure_session(request.couch_user):
```

Comment:
@millerdev I could be reading this wrong, but I think this change removes the part of the workflow where if a superuser visits a secure domain, they also have their session timeout after 30 minutes. I do think we want to bring this back as per the note from @krsdimagi. So the check should be:
```python
_user_requires_secure_session() or _session_insecure_and_domain_requires_secure_session()
```